src/formatDate.js

```diff
@@ -8,7 +8,52 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const fromSeparator = fromFormat.pop();
+  const toSeparator = toFormat.pop();
+
+  const dateArray = date.split(fromSeparator);
+  const modifiedDate = [];
+
+  const fromFormatLength = fromFormat.join('').length;
+  const toFormatLength = toFormat.join('').length;
+
+  if (fromFormatLength > toFormatLength) {
+    const fromYearIndex = fromFormat.indexOf('YYYY');
+    const splitArrayDate = dateArray[fromYearIndex].split('');
+
+    for (let i = 1; i < 3; i++) {
+      splitArrayDate.shift();
+    }
+
+    dateArray[fromYearIndex] = splitArrayDate.join('');
+
+    fromFormat[fromYearIndex] = 'YY';
+  } else if (fromFormatLength < toFormatLength) {
+    const fromYearIndex = fromFormat.indexOf('YY');
+    const splitArrayDate = dateArray[fromYearIndex].split('');
+
+    if (dateArray[fromYearIndex] < 30) {
+      splitArrayDate.unshift(20);
+    } else if (dateArray[fromYearIndex] >= 30) {
+      splitArrayDate.unshift(19);
```

Comment:
The logic for determining the century is hardcoded to assume that years less than 30 belong to the 2000s and years 30 or greater belong to the 1900s. This logic might not be suitable for all scenarios. Consider making this logic configurable or more robust to handle different use cases.